registry/nacos/mcpserver/watcher.go

```diff
@@ -0,0 +1,1010 @@
+	argsData, err := json.Marshal(args)
+	if err != nil {
+		return nil, err
+	}
+	toolArgs := &ToolArgs{}
+	if err = json.Unmarshal(argsData, toolArgs); err != nil {
+		return nil, err
+	}
+	return toolArgs, nil
+}
+
+func getArgsPositionFromToolMeta(toolMeta *ToolsMeta) (map[string]string, error) {
+	result := map[string]string{}
+	if toolMeta == nil {
+		return result, nil
+	}
+	toolTemplate := toolMeta.Templates
+	for kind, meta := range toolTemplate {
+		switch kind {
+		case JsonGoTemplateType:
+			templateData, err := json.Marshal(meta)
+			if err != nil {
+				return result, err
+			}
+			template := &JsonGoTemplate{}
+			if err = json.Unmarshal(templateData, template); err != nil {
+				return result, err
+			}
+			result = mergeMaps(result, template.ArgsPosition)
+		default:
+			return result, fmt.Errorf("unsupport tool meta type %v", kind)
+		}
+	}
+	return result, nil
+}
+
+func getRequestTemplateFromToolMeta(toolMeta *ToolsMeta) (*RequestTemplate, error) {
+	if toolMeta == nil {
+		return nil, nil
+	}
+	toolTemplate := toolMeta.Templates
+	for kind, meta := range toolTemplate {
+		switch kind {
+		case JsonGoTemplateType:
+			templateData, err := json.Marshal(meta)
+			if err != nil {
+				return nil, err
+			}
+			template := &JsonGoTemplate{}
+			if err = json.Unmarshal(templateData, template); err != nil {
+				return nil, err
+			}
+			return &template.RequestTemplate, nil
+		default:
+			return nil, fmt.Errorf("unsupport tool meta type")
+		}
+	}
+	return nil, nil
+}
+
+func getResponseTemplateFromToolMeta(toolMeta *ToolsMeta) (*ResponseTemplate, error) {
+	if toolMeta == nil {
+		return nil, nil
+	}
+	toolTemplate := toolMeta.Templates
+	for kind, meta := range toolTemplate {
+		switch kind {
+		case JsonGoTemplateType:
+			templateData, err := json.Marshal(meta)
+			if err != nil {
+				return nil, err
+			}
+			template := &JsonGoTemplate{}
+			if err = json.Unmarshal(templateData, template); err != nil {
+				return nil, err
+			}
+			return &template.ResponseTemplate, nil
+		default:
+			return nil, fmt.Errorf("unsupport tool meta type")
+		}
+	}
+	return nil, nil
+}
+
+func mergeMaps(maps ...map[string]string) map[string]string {
+	if len(maps) == 0 {
+		return nil
+	}
+	res := make(map[string]string, len(maps[0]))
+	for _, m := range maps {
+		for k, v := range m {
+			res[k] = v
+		}
+	}
+	return res
+}
+
+func (w *watcher) constructWasmFromMcpRule(cfg *WasmPluginConfig, group, dataId string) (*config.Config, error) {
+	rulesBytes, err := json.Marshal(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("marshal rule error:%v", err)
+	}
+	pbs := &structpb.Struct{}
+	if err = protojson.Unmarshal(rulesBytes, pbs); err != nil {
+		return nil, fmt.Errorf("unmarshal rule error:%v", err)
+	}
+
+	wasmPlugin := &extensions.WasmPlugin{
+		ImagePullPolicy: extensions.PullPolicy_Always,
+		Phase:           extensions.PluginPhase_UNSPECIFIED_PHASE,
+		Priority:        &wrapperspb.Int32Value{Value: 30},
+		PluginConfig:    pbs,
```

Comment:
The plugin configuration needs to take effect at the route level; otherwise, it will affect all global routes.